yt/geometry/particle_geometry_handler.py

```diff
@@ -173,6 +173,9 @@ def _initialize_index(self):
         try:
             if dont_load:
                 raise OSError
+            # We must make sure the index order is correct, here and
+            # in the particle bitmap
+            self.ds.index_order = self.regions.correct_index_order(fname)
```

Comment:
Taking into account my comment below, I think this should be rewritten as:
```suggestion
            self.ds.index_order = (self.regions.index_order1, self.regions.index_order2)
```